### PR TITLE
AggLayer integration

### PR DESCRIPTION
**Pull request by vcastellm**
This PR adds config options and logic to send proofs to AggLayer

### CODE REVIEW

File: cmd/run.go [hunk 1]
```diff
@@ -497,7 +498,19 @@ func createSequenceSender(cfg config.Config, pool *pool.Pool, etmStorage *ethtxm
 }
 
 func runAggregator(ctx context.Context, c aggregator.Config, etherman *etherman.Client, ethTxManager *ethtxmanager.Client, st *state.State) {
-	agg, err := aggregator.New(c, st, ethTxManager, etherman)
+	var beethCli *agglayerClient.Client
```
Comment (arnaubennassar) [inline cmd/run.go]:
```suggestion
	var aggCli *agglayerClient.Client
```

File: cmd/run.go [hunk 2]
```diff
@@ -497,7 +498,19 @@ func createSequenceSender(cfg config.Config, pool *pool.Pool, etmStorage *ethtxm
...
+
+	if c.SettlementBackend == aggregator.AggLayer {
+		beethCli = agglayerClient.New(c.AggLayerURL)
+	}
+
+	// Load private key
+	pk, err := config.NewKeyFromKeystore(c.SequencerPrivateKey)
```
Comment (arnaubennassar) [inline cmd/run.go]: only load pk if using aggLayer

File: config/config.go [hunk 3]
```diff
@@ -183,3 +187,19 @@ func Load(ctx *cli.Context, loadNetworkConfig bool) (*Config, error) {
 	}
 	return cfg, nil
 }
+
+// NewKeyFromKeystore creates a private key from a keystore file
+func NewKeyFromKeystore(cfg types.KeystoreFileConfig) (*ecdsa.PrivateKey, error) {
```
Comment (arnaubennassar) [inline config/config.go:192]: nice to have this here, later we can get rid of the old way of getting this through etherman 😄

**Review by arnaubennassar — CHANGES_REQUESTED** [review]
fix the linter, and minor suiggestions

File: cmd/run.go (hunk 1 above)
Comment (vcastellm) [reply cmd/run.go]: good catch

File: cmd/run.go (hunk 2 above)
Comment (vcastellm) [reply cmd/run.go]: Fixed in https://github.com/0xPolygon/cdk-validium-node/pull/71/commits/e94685a03c767922cab92966633b43d0424a2186

**Review by arnaubennassar — APPROVED** [review]

**Review by goran-ethernal — APPROVED** [review]
Looks good to me. Linter errors are left, and I think, `node-config-schema.json` needs to be changed because of the added `SettlementBackend`.

**Pull request closed without merging**